### ISSUE TITLE
Rule idea: dollar-variable-empty-line-before

### ISSUE THREAD

**Issue by jeddy3** [kristerkari/stylelint-scss#53]
- Name: `dollar-variable-empty-line-before`
- Primary option: `"always"|"never"`
- Secondary options: `{ "except": ["first-nested", "after-comment", "after-dollar-variable"] }`
- Messages: 
  - Rejected: "Unexpected empty line before dollar-variable"
  - Expected: "Expected empty line before dollar-variable"
- Section: "`$`-variable"

Given:

``` js
[ "always", { "except": ["first-nested", "after-comment", "after-declaration"] }]
```

The following is valid:

``` css
a {
  $var1: value; /* "always" except first-nested */
  $var2: value; /* "always" except after-declaration */

  /* comment */
  $var3: value; /* "always" except after-comment */

  @extends "x";

  $var4: value; /* primary option is "always" */

   & b {
     prop: value;
   }

   $var5: value;  /* primary option is "always" */
}
```

Given:

``` js
"never"
```

The following is valid:

``` css
a {
  $var: value; /* primary option is "never" */
  $var: value; /* primary option is "never" */

  /* comment */
  $var: value; /* primary option is "never" */

  @extends "x";
  $var: value; /* primary option is "never" */

   & b {
     prop: value;
   }
   $var: value; /* primary option is "never" */
}
```

``` console
1:1 Unexpected empty line before dollar variable (scss/dollar-variable-empty-line-before)
```

Ref: https://github.com/stylelint/stylelint/issues/1573

**Comment by jeddy3** [comment]
FYI, for someone wanting to implement this, a very similar rule, but for custom properties, can be found [here](https://github.com/stylelint/stylelint/tree/master/src/rules/custom-property-empty-line-before).

**Comment by dryoma** [comment]
Added in #85 (at last :) )

**Comment by dryoma** [comment]
Available in 1.4.1
